pkg/infra/initiator_test.go

```diff
@@ -0,0 +1,111 @@
+package infra_test
+
+import (
+	"io/ioutil"
+	"os"
+
+	"tape/e2e"
+	"tape/pkg/infra"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	log "github.com/sirupsen/logrus"
+)
+
+var _ = Describe("Initiator", func() {
+
+	var (
+		configFile *os.File
+		tmpDir     string
+	)
+
+	BeforeEach(func() {
+		mtlsCertFile, err := ioutil.TempFile(tmpDir, "mtls-*.crt")
+		Expect(err).NotTo(HaveOccurred())
+
+		mtlsKeyFile, err := ioutil.TempFile(tmpDir, "mtls-*.key")
+		Expect(err).NotTo(HaveOccurred())
+
+		err = e2e.GenerateCertAndKeys(mtlsKeyFile, mtlsCertFile)
+		Expect(err).NotTo(HaveOccurred())
+
+		mtlsCertFile.Close()
+		mtlsKeyFile.Close()
+
+		configFile, err = ioutil.TempFile(tmpDir, "config*.yaml")
+		Expect(err).NotTo(HaveOccurred())
+		configValue := e2e.Values{
+			PrivSk:   mtlsKeyFile.Name(),
+			SignCert: mtlsCertFile.Name(),
+			Mtls:     false,
+			Addr:     "dummy",
+		}
+		e2e.GenerateConfigFile(configFile.Name(), configValue)
+	})
+
+	AfterEach(func() {
+		os.RemoveAll(tmpDir)
+	})
+
+	It("should crete proposal to raw without limit when limit is 0", func() {
+		logger := log.New()
+
+		raw := make(chan *infra.Elements, 1002)
+		defer close(raw)
+		errorCh := make(chan error, 1002)
+		defer close(errorCh)
+		config, err := infra.LoadConfig(configFile.Name())
+		Expect(err).NotTo(HaveOccurred())
+		crypto, err := config.LoadCrypto()
+		Expect(err).NotTo(HaveOccurred())
+		infra.StartCreateProposal(1002, 10, 0, config, crypto, raw, errorCh, logger)
+		Expect(len(raw)).Should(Equal(1002))
+		<-raw
+	})
+
+	It("should crete proposal to raw without limit when limit less than 0", func() {
```

Comment:
could we simply disallow a negative limit?

Reply:
Will update.